compiler/src/iree/compiler/InputConversion/Common/ImportMLProgram.cpp

```diff
@@ -77,9 +82,86 @@ class MLProgramGlobalOpPattern
         srcOpAttr.has_value()
             ? Optional<TypedAttr>(srcOpAttr.value().cast<TypedAttr>())
             : llvm::None;
+    const bool isMutable = srcOp.getIsMutable();
+    const SymbolTable::Visibility visibility = srcOp.getVisibility();
     auto globalOp = rewriter.replaceOpWithNewOp<IREE::Util::GlobalOp>(
         srcOp, srcOp.getName(), srcOp.getIsMutable(), newType, srcOpTypedAttr);
-    globalOp.setVisibility(srcOp.getVisibility());
+    globalOp.setVisibility(visibility);
+
+    if (visibility != SymbolTable::Visibility::Public) return success();
+
+    ModuleOp module = srcOp->getParentOfType<ModuleOp>();
+
+    // Default get & set.
+    std::string getterName, setterName;
+
+    auto verifyFormat = [](const std::string &format) {
```

Comment:
can you add a note about what this is doing? (a few expected input/output examples are probably enough)